editor/scene_tree_dock.cpp

```diff
@@ -1032,51 +1032,119 @@ void SceneTreeDock::_tool_selected(int p_tool, bool p_confirm_override) {
 			}
 
 			List<Node *> selection = editor_selection->get_selected_node_list();
-			List<Node *>::Element *e = selection.front();
-			if (e) {
-				Node *node = e->get();
-				if (node) {
-					bool editable = EditorNode::get_singleton()->get_edited_scene()->is_editable_instance(node);
 
-					if (editable) {
-						editable_instance_remove_dialog->set_text(TTR("Disabling \"editable_instance\" will cause all properties of the node to be reverted to their default."));
-						editable_instance_remove_dialog->popup_centered();
-						break;
+			if (selection.size() > 1) {
+				// Check if the selection contains both editable and non-editable nodes.
+				bool has_editables = false;
+				bool has_non_editables = false;
+				for (const Node *node : selection) {
+					if (node) {
+						bool editable = EditorNode::get_singleton()->get_edited_scene()->is_editable_instance(node);
+						has_editables |= editable;
+						has_non_editables |= !editable;
+						if (has_non_editables && has_editables) {
+							break;
+						}
 					}
+				}
+
+				// Fire dialog if selection does contain both, to ask user what they want to do (enable or disable).
+				if (has_non_editables && has_editables) {
+					editable_instance_enable_or_disable_dialog->popup_centered();
+					break;
+				}
+
+				if (has_editables) {
+					dialog_option_enable_all = false;
+					editable_instance_remove_dialog->set_text(TTR("Disabling \"Editable Children\" will cause all properties of those nodes to be reverted to their default."));
```

Comment:
```suggestion
					editable_instance_remove_dialog->set_text(TTR("Disabling \"Editable Children\" will cause all properties of selected nodes to be reverted to their default."));
```

Seems like you've copied this text from another context where it was clear which are "those" nodes, but here it isn't.